src/UI/Game.jsx

```diff
@@ -0,0 +1,458 @@
+import moveSound from '../assets/audio/move.wav';
+
+var keyPressAllow = true;
+const secret = 'secret';
+class Game extends Component {
+  constructor(props) {
+    super(props);
+    this.state = this.getAllData();
+    this.signIn = this.signIn.bind(this);
+    this.signUp = this.signUp.bind(this);
+    this.modal = this.modal.bind(this);
+    this.register = this.register.bind(this);
+    this.login = this.login.bind(this);
+    this.exitModal = this.exitModal.bind(this);
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+    this.gameOver = this.gameOver.bind(this)
+    this.moveSound = new Audio(moveSound);
+  }
+
+  getAllData() {
+    const bestScore = localStorage.getItem("bestScore") ||  0;
+    const mainSettings = localStorage.getItem("s") ? JSON.parse(cryptojs.AES.decrypt(localStorage.getItem("s"), secret).toString(cryptojs.enc.Utf8))
+    : {
+      cells: initCells(4),
+      score: 0,
+      size: 4,
+    };
+    this.props.updateField(mainSettings.cells);
+    this.props.addScore(mainSettings.score);
+    this.props.setSize(mainSettings.size);
+    this.props.setSoundVolume(localStorage.getItem("soundVolume") ||  0.5);
+    return {
+      additionalScore: 0,
+      bestScore,
+      modal: {
+        active: false,
+        data: {
+          type: 'error',
+          exit: true,
+          icon: error,
+          title: "Error",
+          description: "An error occurred",
+          inputId: "input",
+          error: null,
+          firstButton: {
+            text: "OK",
+            handler: null
+          },
+          secondButton: {
+            active: false,
+            text: "",
+            handler: null
+          },
+          submit: ''
+        }
+      }
+    }
+  }
+
+  saveAllData(state) {
+    const cells = this.props.cells;
+    const score = this.props.score;
+    const size = this.props.size;
+    localStorage.setItem("bestScore", state.bestScore);
+    localStorage.setItem("s", cryptojs.AES.encrypt(JSON.stringify({cells, score, size}), secret).toString());
+    localStorage.setItem("nightmode", this.props.nightmode);
+  }
+
+  exitModal() {
+    keyPressAllow = true;
+    this.setState((state) => state.modal.active = false);
+  }
+
+  signIn(exit) {
+    this.modal(true, "login", {
+      exit: exit !== undefined ? exit : true,
+      firstButton: {
+        text: "Login",
+        handler: (e) => e.preventDefault()
+      },
+      secondButton: {
+        text: "Register",
+        handler: () => this.signUp(exit),
+      },
+      submit: this.login
+    })
+  }
+
+  signUp(exit) {
+    this.modal(true, "register", {
+      exit: exit !== undefined ? exit : true,
+      firstButton: {
+        text: "Register",
+        handler: (e) => e.preventDefault()
+      },
+      secondButton: {
+        text: "Login",
+        handler: () => this.signIn(exit),
+      },
+      submit: this.register
+    })
+  }
+
+  register(event, data) {
+    if (data.password !== data.confirm) {
```

Comment:
You should use two if-statements instead of long if-elses for better readability: one for username, one for password